src/streamlit_app/app.py

```diff
@@ -63,27 +70,77 @@ def main():
         value=dv.model_confidence_value,
     )
     # Type of device to use, i.e. CPU or GPU
-    #
     st.sidebar.markdown("---")
     # -- Media-specific
     st.sidebar.subheader("Media configuration")
     # Type of media to use, i.e. video or image
     input_data_type = st.sidebar.radio(
-        "Select input type: ",
-        ["image", "video"],
+        "Select media type: ",
+        [MediaType.VIDEO],
     )
     # Input source, i.e. sample data, local, or on the web.
     data_src = st.sidebar.radio(
-        "Select input source: ",
+        "Select media source: ",
         [
-            "Sample data",
-            "URL of the media",
-            "Upload your own data",
+            MediaSource.SAMPLE_DATA,
+            MediaSource.URL,
         ],
     )
-    logger.info(f"{data_src} | {input_data_type} | {confidence}")
+    logger.info(f"Data Source: {data_src}")
+    logger.info(f"Type: {input_data_type}")
+    logger.info(f"Confidence: {confidence}")
+
+    return data_src
+
+
+def get_media_url(data_src: str) -> str:
+    if data_src == MediaSource.URL:
+        return st.sidebar.text_input("URL")
+    else:
+        return dv.video_url
+
+
+def handle_analysis(url: str):
+    if not url:
+        st.warning("Please enter a URL for you meida.")
+        return
+
+    st.markdown(f"The target URL is: {url}")
+
+    try:
+        video_data = VideoData(url=url)
+    except Exception as exception:
+        st.warning(
+            f"There was a problem processing the URL.\n\nError: {exception}"
+        )
+        return
+
+    height = 0
+    width = 0
+    fps = 0
+    st1, st2, st3 = st.columns(3)
+    with st1:
+        st.markdown("## Height")
+        st1_text = st.markdown(f"{height}")
+    with st2:
+        st.markdown("## Width")
+        st2_text = st.markdown(f"{width}")
+    with st3:
+        st.markdown("## FPS")
+        st3_text = st.markdown(f"{fps}")
+
+    st.markdown("---")
+    output = st.empty()
 
-    return
+    # TODO Replace temp method with analysis workflow
+    results = run_image_analysis(video_data)
+    for frame in results:
+        logger.info("Frame results: ", frame)
+        output.image(frame.output_image)
+        fps = frame.fps
```

Comment:
I think that `fps` was calculated as the difference between `prev_time` and `curr_time`, but I'd have to check. If the `fps` is the same for all frames, we can simply have it be a default value.

Reply:
So my thought was that during the processing we'd calculate the `fps` from the `prev_time` and `curr_time` like you're thinking from the example. That's actually done in the loop for each frame. Check out [line 85 here](https://github.com/moaaztaha/Yolo-Interface-using-Streamlit/blob/main/app.py#L85). When we "replay" the collected results we can display this value on the UI. I'm good with whatever we decide, I just added it in since it was in our example.

Reply:
This makes sense. We can update this once we have the rest of the code from the Yolo portion.